### ISSUE TITLE
Return `seq[ContactsDetails]` from `getContactsByGroup`

### ISSUE THREAD

**Issue by osmaczko** [status-im/status-desktop#10807]
TLDR: the `getContactsByGroup` function in `src/app_service/service/contacts/service.nim` currently returns `seq[ContactsDto]`, but that return value is most of the time used to get `ContactsDetails` and the original property is already `ContactsDetails`, so we're doing type transfers uselessly. So change the signature to return `seq[ContactsDetails]` directly

              Maybe return `seq[ContactsDetails]` from this function?

_Originally posted by @igor-sirotin in https://github.com/status-im/status-desktop/pull/10805#discussion_r1205503613_

              This function is called with `getContacts` in `initContactRequestsModel`.
              And then for each contact we call `createItemFromPublicKey` which calls `getContactDetails`.

              So we do have `ContactDetails` in the beginning, then switch it to `ContactDto` and eventually back to `ContactDetails`.

_Originally posted by @igor-sirotin in https://github.com/status-im/status-desktop/pull/10805#discussion_r1205535598_
